### ISSUE TITLE
`Dendron: Migrate to self contained vault` does not migrate the vault correctly

### ISSUE THREAD

**Issue by ccamara** [dendronhq/dendron#3466]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [x] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [x] Lookup.  Dendron's Lookup Command
- [ ] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [ ] Publish.  External Site Publish
- [ ] Markdown.  Markdown features, such as syntax support and features

## Describe the bug

Whenever I add a remote, self-contained vault to my workspace I get an error such as the one below + the vault is not recognized by dentron (Lookup won't work + not displayed in dendron's tab).

```
{"fsPath":"cim-shared-vault","remote":{"type":"git","url":"git@<actual_git_repo>.git"},"name":"CIM Shared vault","selfContained":true}
```

The graph panel would show this:

```
{"name":"DendronError","status":"unknown","payload":"No Engine: <path>/Dendron/cim-dendron"}
```

## To Reproduce

1. Add a remote vault from existing dendron workspace
2. Convert the vault to self-contained

## Expected behavior

I would expect dendron to work with that vault, like it does with the other not-self contained.

## Desktop (please complete the following information)

- OS: macOS 12.5
- Browser (if with published websites; include version): [e.g. chrome, safari]
- Versions (all that apply to bug):
  - Dendron Extension: v0.109.1
  - VS Code version: 1.70.2
  - Dendron CLI:

## Dendron Log file

Please attach the [output of `> Dendron:Dev: Open Logs` here](https://cloud.montera34.org/index.php/s/2MNqFwEfZXWZ6Yy)

**Comment by Harshita-mindfire** [comment]
Hey @ccamara , Thanks for reporting the bug.
Can you elaborate on the steps you proceeded? 
I tried to add this remote self contained vault : https://github.com/Harshita-mindfire/test-vault into my workspace and everything works fine. Is there a step I'm missing?

**Comment by ccamara** [comment]
Sure!

I tested something simpler:

1. Open my dendron workspace
2. Create a new local vault: `Dendron: Vault Add\Local`
3. Generate a new vault
4. Create a new note
5. The vault is added properly and works well
6. Migrate the vault created in 2: `Dendron: Migrate to self contained vault`

Multiple errors: On Dendron's tab:

1. On graph note, I get the following error: `{"name":"DendronError","status":"unknown","payload":"No Engine: /Users/carloscamara/Documents/Dendron/cim-dendron"}`
2. Tree view is empty
3. Lookup won't find anything on the new vault

**Comment by Harshita-mindfire** [comment]
got it. This should be the error with `Dendron: Migrate to self contained vault` command. Let me update the bug title.

**Comment by ccamara** [comment]
OK, I believe I may know what the problem might be.

Whenever I migrate to a self contained vault, all notes are moved to `/note/note` (please, note the note folder is added twice).
Moving contents from `/note/note` to `/note` fixes the problem.

**Comment by Harshita-mindfire** [comment]
A followup ques: Do you have `enableSelfContainedVaults` set as `false` in your dendron.yml?
```yml
dev:
    enableSelfContainedVaults: false
```
Presently all the new vault that are added are by-default `self-contained`.

**Comment by ccamara** [comment]
That might be it: there's no mention to `enableSelfContainedVaults:` in my dendron.yml

**Comment by Harshita-mindfire** [comment]
Yeah, since it is a new vault and already self contained, It should work just fine.

**Comment by ccamara** [comment]
No, I can confirm that if I migrate an existing vault into a self-contained vault, dendron breaks.

Steps:

1. Open a workspace with several vaults
2. Migrate local vault to self-contained (Dendron: Migrate to self contained vault)


Multiple errors: On Dendron's tab:

    On graph note, I get the following error: {"name":"DendronError","status":"unknown","payload":"No Engine: /Users/carloscamara/Documents/Dendron/cim-dendron"}
    Tree view is empty
    Lookup won't find anything on the new vault

**Comment by Harshita-mindfire** [comment]
yeah I can repro this error with `Dendron: Migrate to self contained vault`, hence updated the title. We will inform you when this issue gets fixed. 🙏

**Comment by ccamara** [comment]
Thanks, @Harshita-mindfire !

**Comment by ccamara** [comment]
Hi, I've seen the PR has been merged, although not sure if it has been implemented already. This is just to update that When running `Dendron: Migrate to self contained vault` in Dendron v0.111.0, the problem still persists.

Just FYI, I can see notes moved to `notes/` folder (as opposed to what happened before: `notes/notes`) but `root.md` and `root.schema.yml` are located in `notes/` . When I manually move those files to the root of the vault, dendron works just fine.

**Comment by Harshita-mindfire** [comment]
Hey @ccamara this PR is merged, but will release in the upcoming release i.e. V0.112

**Comment by ccamara** [comment]
Great news, thanks @Harshita-mindfire !

**Comment by Harshita-mindfire** [comment]
This should be fixed with Dendron v0.112.0. Feel free to ping here incase of any issues

**Comment by ccamara** [comment]
I can confirm that this works as expected in Dendron v.0.112.0. Thank you so much!
